crates/optimism/payload/src/builder.rs

```diff
@@ -220,6 +220,26 @@ where
             None,
         ))
     }
+
+    fn cfg_and_block_env(
+        &self,
+        chain_spec: &ChainSpec,
+        parent: &Header,
+    ) -> (CfgEnvWithHandlerCfg, BlockEnv) {
+        let mut cfg_env = CfgEnvWithHandlerCfg::new_with_spec_id(CfgEnv::default(), SpecId::LATEST);
+        let mut block_env = BlockEnv::default();
```

Comment:
please use the `BlockEnv` we used in optimism prior to this change, eg with `number` equal to `parent.number + 1` and so on

Reply:
no sorry, you are correct, I forgot that `fill_cfg_and_block_env` mutates the passed in `block_env`